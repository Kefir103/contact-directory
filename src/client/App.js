import React from 'react';
import { connect } from 'react-redux';
import Table from './components/table/Table';
import SideMenu from './components/sidemenu/SideMenu';
import Header from './components/Header';
import AddEntryComponent from './components/entry/AddEntryComponent';
import EntryInfo from './components/entry/EntryInfo';
import { Error } from './components/Error';
import Loader from './components/Loader';

const App = (props) => {
    return (
        <>
            <Header />
            <SideMenu />
            {!props.error ? (
                <div className={'app-container'}>
                    {!props.isLoading ? (
                        [
                            <AddEntryComponent key={'addEntryComponent'} />,
                            <Table key={'table'} />,
                            <div className={'entry-info-container'} key={'entryInfoContainer'}>
                                {props.currentElement.elementInfo ? <EntryInfo /> : ''}
                            </div>,
                        ]
                    ) : (
                        <Loader />
                    )}
                </div>
            ) : (
                <Error />
            )}
        </>
    );
};

const mapStateToProps = (state) => {
    return {
        isLoading: state.appStatus.isLoading,
        currentElement: state.data.currentElement,
        error: state.appStatus.error,
    };
};

export default connect(mapStateToProps)(App);
