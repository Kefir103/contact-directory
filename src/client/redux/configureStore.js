import { createStore, applyMiddleware, compose } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/index';
import { rootReducer } from './reducers/rootReducer';
import thunk from 'redux-thunk';

let store;

export default function configureStore(initialState) {
    if (store) {
        return store;
    }

    store =
        process.env.NODE_ENV === 'production'
            ? createStore(rootReducer, initialState, compose(applyMiddleware(thunk)))
            : createStore(rootReducer, initialState, composeWithDevTools(applyMiddleware(thunk)));

    return store;
}
