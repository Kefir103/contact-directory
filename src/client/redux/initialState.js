export const initialState = {
    filter: {
        sortingMap: new Map([]),
        filterText: '',
        filterFields: [],
        currentPage: 1,
    },
    data: {
        currentElement: {
            elementInfo: undefined,
            elementIndex: -1,
        },
        elements: [],
        currentElements: [],
        filteredElements: undefined,
        inputContainer: {
            isAddButtonDisabled: false,
            isFormOpen: false,
            inputElement: {
                id: 0,
                firstName: '',
                lastName: '',
                email: '',
                phone: '',
            },
            validInputs: {
                id: false,
                firstName: false,
                lastName: false,
                email: false,
                phone: false,
            },
            addingStatus: {
                addingStatusText: '',
                isError: false,
            },
        },
    },
    appStatus: {
        isLoading: false,
        error: undefined,
    },
};
