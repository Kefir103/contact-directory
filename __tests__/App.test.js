import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { initialState } from '../src/client/redux/initialState';
import App from '../src/client/App';

const mockStore = configureMockStore();

describe('<App />', () => {
    test('should render correctly with error', () => {
        const store = mockStore({
            ...initialState,
            appStatus: {
                ...initialState.appStatus,
                error: true,
            },
        });
        const wrapper = shallow(<App store={store} />);
        const component = wrapper.dive().dive();
        expect(component).toMatchSnapshot();
    });
    test('should render correctly with isLoading', () => {
        const store = mockStore({
            ...initialState,
            appStatus: {
                ...initialState.appStatus,
                isLoading: true,
            },
        });
        const wrapper = shallow(<App store={store} />);
        const component = wrapper.dive().dive();
        expect(component).toMatchSnapshot();
    });
    test('should render correctly without isLoading and elementInfo', () => {
        const store = mockStore({ ...initialState });
        const wrapper = shallow(<App store={store} />);
        const component = wrapper.dive().dive();
        expect(component).toMatchSnapshot();
    });
    test('should render correctly without isLoading and with elementInfo', () => {
        const store = mockStore({
            ...initialState,
            data: {
                ...initialState.data,
                currentElement: {
                    elementInfo: 'elementInfo',
                    elementIndex: 1,
                },
            },
        });
        const wrapper = shallow(<App store={store} />);
        const component = wrapper.dive().dive();
        expect(component).toMatchSnapshot();
    });
});
