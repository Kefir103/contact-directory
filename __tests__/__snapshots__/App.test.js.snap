// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly with error 1`] = `
<Fragment>
  <Header />
  <Connect(SideMenu) />
  <Error />
</Fragment>
`;

exports[`<App /> should render correctly with isLoading 1`] = `
<Fragment>
  <Header />
  <Connect(SideMenu) />
  <div
    className="app-container"
  >
    <Loader />
  </div>
</Fragment>
`;

exports[`<App /> should render correctly without isLoading and elementInfo 1`] = `
<Fragment>
  <Header />
  <Connect(SideMenu) />
  <div
    className="app-container"
  >
    <Connect(AddEntryComponent) />
    <Connect(Table) />
  </div>
</Fragment>
`;

exports[`<App /> should render correctly without isLoading and with elementInfo 1`] = `
<Fragment>
  <Header />
  <Connect(SideMenu) />
  <div
    className="app-container"
  >
    <Connect(AddEntryComponent) />
    <Connect(Table) />
    <Connect(EntryInfo) />
  </div>
</Fragment>
`;
